src/tests/zf_apps/zfsink.c

```diff
@@ -21,10 +21,23 @@
 #include <stdbool.h>
 #include <stdarg.h>
 #include <inttypes.h>
+#include <pthread.h>
+#include <sys/time.h>
+
+
+struct resources {
+  /* statistics */
+  volatile uint64_t n_rx_pkts;
+  volatile uint64_t n_rx_bytes;
+};
 
 
 static bool cfg_quiet = false;
 static bool cfg_rx_timestamping = false;
+static struct resources res;
```

Comment:
minor comment - I would use

`static struct resources res = {0};`

here and save the lines initialising later in `main()`

Reply:
Objects with `static` storage duration will [implicitly](https://en.cppreference.com/w/c/language/initialization#:~:text=objects%20with%20static%20and%20thread%2Dlocal%20storage%20duration%20are%20empty%2Dinitialized) be [empty-initialised](https://en.cppreference.com/w/c/language/initialization#:~:text=pointers%20are%20initialized,initialized%20to%20zero), so the current declaration is sufficient to set the values to 0.